src/add-user/add-user.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useState} from 'react';
+import {Alert, Button, Text, TextInput, View} from 'react-native';
+import {
+  validateBirthDate,
+  validateEmail,
+  validateName,
+  validatePassword,
+  validatePhone,
+} from '../utils/validation';
+
+export const AddUserScreen = () => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [phone, setPhone] = useState('');
+  const [birthDate, setBirthDate] = useState('');
+
+  const validateUserData = () => {
+    const emailValidation = validateEmail(email);
+    const passwordValidation = validatePassword(password);
+    const phoneValidation = validatePhone(phone);
+    const nameValidation = validateName(name);
+    const birthDateValidation = validateBirthDate(birthDate);
+    if (emailValidation !== '') {
+      return Alert.alert(emailValidation);
+    }
+
+    if (passwordValidation !== '') {
```

Comment:
como as strings podem ser `undefined` (já que estão no `useState`), o que acha de algo assim?
```suggestion
    if (!passwordValidation?.length) {
```